### ISSUE TITLE
Track2 request break down

### ISSUE THREAD

**Issue by yupwei68** [Azure/azure-sdk-for-go#14598]
### Bug Report

- import path of package in question, e.g. `.../services/compute/mgmt/2018-06-01/compute`
- SDK version e.g. `master`, `latest`, `18.1.0`
  - Specify the exact commit if possible; one way to get this is the REVISION
    column output by `dep status "github.com/Azure/azure-sdk-for-go`.
- output of `go version`

- What happened?
The polling call for creation of an AVS private cloud breaks down

- What did you expect or want to happen?
The polling call for creation of an AVS private cloud succeeds

- How can we reproduce it?
calling https://github.com/ArcturusZhang/azure-sdk-for-go/blob/avs-track2/sdk/arm/avs/2020-03-20/armavs/zz_generated_models.go#L595

It causes panic in type convertion, here https://github.com/ArcturusZhang/azure-sdk-for-go/blob/avs-track2/sdk/azcore/request.go#L238 
log:
```
panic: interface conversion: http.noBody is not io.Seeker: missing method Seek

goroutine 133 [running]:
github.com/Azure/azure-sdk-for-go/sdk/azcore.(*Request).RewindBody(...)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/azcore/request.go:199
github.com/Azure/azure-sdk-for-go/sdk/azcore.(*retryPolicy).Do(0xc00011b200, 0xc001161cb0, 0x0, 0x5e6fc00, 0xc001300ff0)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/azcore/policy_retry.go:157 +0xd25
github.com/Azure/azure-sdk-for-go/sdk/azcore.(*Request).Next(0xc001161c80, 0x0, 0xc0011befd8, 0x61779f)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/azcore/request.go:112 +0xdc
github.com/Azure/azure-sdk-for-go/sdk/armcore.(*rpRegistrationPolicy).Do(0xc001374300, 0xc001161c80, 0xc0011bf100, 0x97afdbb77a328e06, 0x10)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/armcore/policy_register_rp.go:117 +0x4a5
github.com/Azure/azure-sdk-for-go/sdk/azcore.(*Request).Next(0xc001161c50, 0xc001161bf0, 0x5810030, 0xa)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/azcore/request.go:112 +0xdc
github.com/Azure/azure-sdk-for-go/sdk/azcore.telemetryPolicy.Do(0xc0003dc000, 0xca, 0xc001161c50, 0x562eea0, 0x1, 0xc001161c50)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/azcore/policy_telemetry.go:80 +0x1dc
github.com/Azure/azure-sdk-for-go/sdk/azcore.(*Request).Next(0xc001161c20, 0x0, 0x0, 0xc0011bf3bc)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/azcore/request.go:112 +0xdc
github.com/Azure/azure-sdk-for-go/sdk/azcore.Pipeline.Do(0xc0015e4000, 0x9, 0x10, 0xc001161c20, 0xc001810500, 0xf8, 0xc001161c20)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/azcore/core.go:93 +0x74
github.com/Azure/azure-sdk-for-go/sdk/armcore.(*pollingTrackerBase).pollForStatus(0xc00130e6e0, 0x5eded18, 0xc001655740, 0xc0015e4000, 0x9, 0x10, 0x0, 0x0)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/armcore/poller.go:486 +0xec
github.com/Azure/azure-sdk-for-go/sdk/armcore.(*poller).pollForStatus(0xc0011c5090, 0x5eded18, 0xc001655740, 0xc0015e4000, 0x9, 0x10, 0x5fb3100, 0xc00130e6e0, 0xc0010847e0)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/armcore/poller.go:261 +0xaf
github.com/Azure/azure-sdk-for-go/sdk/armcore.(*poller).Poll(0xc0011c5090, 0x5eded18, 0xc001655740, 0xc0015e4000, 0x9, 0x10, 0x1, 0x1, 0x0)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/armcore/poller.go:197 +0x78
github.com/Azure/azure-sdk-for-go/sdk/armcore.(*poller).PollUntilDone(0xc0011c5090, 0x5eded18, 0xc001655740, 0x77359400, 0xc0015e4000, 0x9, 0x10, 0x4e55740, 0xc00180a4c0, 0xc000b021c0, ...)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/armcore/poller.go:222 +0x17e
github.com/Azure/azure-sdk-for-go/sdk/arm/avs/2020-03-20/armavs.(*privateCloudPoller).pollUntilDone(0xc001e240f0, 0x5eded18, 0xc001655740, 0x77359400, 0xc001e240f0, 0xc0015e4000, 0xc0011c5090, 0x5ee5cd8)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/arm/avs/2020-03-20/armavs/zz_generated_pollers.go:182 +0xa4
github.com/Azure/azure-sdk-for-go/sdk/arm/avs/2020-03-20/armavs.(*PrivateCloudsClient).BeginCreateOrUpdate.func1(0x5eded18, 0xc001655740, 0x77359400, 0xc0016205d0, 0x15, 0xc0016328b0, 0xf)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/arm/avs/2020-03-20/armavs/zz_generated_privateclouds_client.go:52 +0x48
github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/vmware.resourceVmwarePrivateCloudCreate(0xc00116fb90, 0x5067dc0, 0xc000380300, 0x0, 0x0)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/vmware/vmware_private_cloud_resource.go:249 +0xb6f
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc0008def30, 0xc0000a6a00, 0xc0011853a0, 0x5067dc0, 0xc000380300, 0x5094a01, 0xc0017b5418, 0xc0010989f0)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:310 +0x375
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000127400, 0xc0011bfa38, 0xc0000a6a00, 0xc0011853a0, 0xc0017b5478, 0xc0017bce50, 0x50971e0)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/provider.go:294 +0x99
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc000287338, 0x5eded50, 0xc001160870, 0xc00116f6c0, 0xc000287338, 0xc001160870, 0xc000080ba0)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:885 +0x8a5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x553b820, 0xc000287338, 0x5eded50, 0xc001160870, 0xc0016554a0, 0x0, 0x5eded50, 0xc001160870, 0xc0017c4500, 0x4a3)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3305 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00067aa80, 0x5f21738, 0xc0003d5680, 0xc000123100, 0xc000b2c090, 0x90b6da0, 0x0, 0x0, 0x0)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1180 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00067aa80, 0x5f21738, 0xc0003d5680, 0xc000123100, 0x0)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1503 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000df0200, 0xc00067aa80, 0x5f21738, 0xc0003d5680, 0xc000123100)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:843 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:841 +0x1fd

```
Error: The terraform-provider-azurerm plugin crashed!

Previously, the request has caused breakdown in io.Seeker conversion in a response for 500 gateway timeout.  
- Anything we should know about your environment.

**Comment by yupwei68** [comment]
Another reproduce log with request and response:

PUT /subscriptions/.../resourceGroups/acctestRG-Vmware-yupp2/providers/Microsoft.AVS/privateClouds/acctest-PC-yupp2?api-version=2020-03-20 HTTP/1.1
 Host: management.azure.com
 User-Agent: azsdk-go-armavs/2020-03-20/v0.1.0 HashiCorp Terraform/0.15.0 (+https://www.terraform.io) Terraform Plugin SDK/1.16.0 terraform-provider-azurerm/dev pid-222c6c49-1b0a-5959-a213-6608f9eb8820 armcore/v0.7.0 azcore/v0.13.4 (go1.16.3; linux)
 Content-Length: 207
 Accept: application/json
 Content-Type: application/json
 Accept-Encoding: gzip
 
 {"location":"australiaeast","tags":{},"properties":{"internet":"Disabled","managementCluster":{"clusterSize":3},"networkBlock":"192.168.48.0/22","nsxtPassword":"","vcenterPassword":""},"sku":{"name":"av36"}}
 [DEBUG] AzureRM Response for https://management.azure.com/subscriptions/.../resourceGroups/acctestRG-Vmware-yupp2/providers/Microsoft.AVS/privateClouds/acctest-PC-yupp2?api-version=2020-03-20: 
 HTTP/2.0 500 Internal Server Error
 Content-Length: 95
 Cache-Control: no-cache
 Content-Type: application/json; charset=utf-8
 Date: Tue, 27 Apr 2021 09:00:50 GMT
 Expires: -1
 Pragma: no-cache
 Server: nginx
 Strict-Transport-Security: max-age=31536000; includeSubDomains
 X-Content-Type-Options: nosniff
 X-Ms-Correlation-Request-Id: 4032bb65-951a-43b2-9530-c29d947a29a2
 X-Ms-Failure-Cause: service
 X-Ms-Ratelimit-Remaining-Subscription-Writes: 1199
 X-Ms-Request-Id: eabec194-b8fb-445c-bdf9-b1d663520f13
 X-Ms-Routing-Request-Id: WESTUS2:20210427T090051Z:4032bb65-951a-43b2-9530-c29d947a29a2
 
 {"error":{"code":"InternalError","message":"Create HCX Subscription or Activation Key error."}}
 PANIC: interface conversion: io.nopCloser is not io.Seeker: missing method Seek
 
 goroutine 110 [running]:
 github.com/Azure/azure-sdk-for-go/sdk/azcore.(*Request).RewindBody(...)
 	/home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/azcore/request.go:199

**Comment by ArcturusZhang** [comment]
Hi @jhendrixMSFT could you please take a look at this?

**Comment by jhendrixMSFT** [comment]
These two issues are slightly different, although I suspect they have the same root cause.  Can you tell me anything else about how the code is used, e.g. is there a custom HTTP transport or anything like that?

**Comment by yupwei68** [comment]
Here I use a `BeginCreateOrUpdate` here https://github.com/ArcturusZhang/azure-sdk-for-go/blob/avs-track2/sdk/arm/avs/2020-03-20/armavs/zz_generated_privateclouds_client.go#L34.
Then I use a `PollUntilDone` here https://github.com/ArcturusZhang/azure-sdk-for-go/blob/avs-track2/sdk/arm/avs/2020-03-20/armavs/zz_generated_models.go#L595.


The code is like:
```
future, err := client.BeginCreateOrUpdate(ctx, resourceGroup, name, privateCloud, nil)
	if err != nil {
		return fmt.Errorf("creating Vmware Private Cloud %q (Resource Group %q): %+v", name, resourceGroup, err)
	}

	if _, err = future.PollUntilDone(ctx, common.DefaultPollingInterval); err != nil {
		return fmt.Errorf("waiting for creation of the Vmware Private Cloud %q (Resource Group %q): %+v", name, resourceGroup, err)
	}
```

**Comment by yupwei68** [comment]
Their panic log is different, the second is as here:

Stack trace from the terraform-provider-azurerm plugin:

```
panic: interface conversion: io.nopCloser is not io.Seeker: missing method Seek

goroutine 110 [running]:
github.com/Azure/azure-sdk-for-go/sdk/azcore.(*Request).RewindBody(...)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/azcore/request.go:199
github.com/Azure/azure-sdk-for-go/sdk/azcore.(*retryPolicy).Do(0xc00027f5c0, 0xc000a0fa70, 0x0, 0x0, 0x0)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/azcore/policy_retry.go:157 +0xd25
github.com/Azure/azure-sdk-for-go/sdk/azcore.(*Request).Next(0xc000a0fa40, 0xca, 0x73, 0xc000f17270)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/azcore/request.go:112 +0xdc
github.com/Azure/azure-sdk-for-go/sdk/armcore.(*rpRegistrationPolicy).Do(0xc001ab80c0, 0xc000a0fa40, 0xc000f17278, 0x2e10876c0bdcfdea, 0x10)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/armcore/policy_register_rp.go:117 +0x4a5
github.com/Azure/azure-sdk-for-go/sdk/azcore.(*Request).Next(0xc000a0fa10, 0xc000a0f320, 0x5810030, 0xa)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/azcore/request.go:112 +0xdc
github.com/Azure/azure-sdk-for-go/sdk/azcore.telemetryPolicy.Do(0xc0014cd680, 0xec, 0xc000a0fa10, 0x562eea0, 0x1, 0xc000a0fa10)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/azcore/policy_telemetry.go:80 +0x1dc
github.com/Azure/azure-sdk-for-go/sdk/azcore.(*Request).Next(0xc000a0f350, 0x0, 0x0, 0xc0011d49e8)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/azcore/request.go:112 +0xdc
github.com/Azure/azure-sdk-for-go/sdk/azcore.Pipeline.Do(0xc0005da300, 0x9, 0x10, 0xc000a0f350, 0x16, 0xc00170e5f0, 0x10)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/azcore/core.go:93 +0x74
github.com/Azure/azure-sdk-for-go/sdk/arm/avs/2020-03-20/armavs.(*PrivateCloudsClient).createOrUpdate(0xc0006a2d08, 0x5eded18, 0xc001710f60, 0xc0017266d8, 0x16, 0xc00170e5f0, 0x10, 0x0, 0x0, 0x0, ...)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/arm/avs/2020-03-20/armavs/zz_generated_privateclouds_client.go:76 +0x125
github.com/Azure/azure-sdk-for-go/sdk/arm/avs/2020-03-20/armavs.(*PrivateCloudsClient).BeginCreateOrUpdate(0xc0006a2d08, 0x5eded18, 0xc001710f60, 0xc0017266d8, 0x16, 0xc00170e5f0, 0x10, 0x0, 0x0, 0x0, ...)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/Azure/azure-sdk-for-go/sdk/arm/avs/2020-03-20/armavs/zz_generated_privateclouds_client.go:35 +0xcf
github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/vmware.resourceVmwarePrivateCloudCreate(0xc0003547e0, 0x5067dc0, 0xc0003be300, 0x0, 0x0)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/vmware/vmware_private_cloud_resource.go:244 +0x9e5
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc001005440, 0xc000317680, 0xc00176cb00, 0x5067dc0, 0xc0003be300, 0x5094a01, 0xc000f02b58, 0xc000f07980)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:310 +0x375
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc0000afc00, 0xc000f17a38, 0xc000317680, 0xc00176cb00, 0xc000f02d78, 0xc000bd4200, 0x50971e0)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/provider.go:294 +0x99
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc000011ac8, 0x5eded50, 0xc000dc7440, 0xc000354310, 0xc000011ac8, 0xc000dc7440, 0xc0016c5ba0)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:885 +0x8a5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x553b820, 0xc000011ac8, 0x5eded50, 0xc000dc7440, 0xc001710cc0, 0x0, 0x5eded50, 0xc000dc7440, 0xc00145aa00, 0x4a7)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3305 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00022ee00, 0x5f21738, 0xc000103e00, 0xc001120c00, 0xc0005d4ae0, 0x90b6da0, 0x0, 0x0, 0x0)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1180 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00022ee00, 0x5f21738, 0xc000103e00, 0xc001120c00, 0x0)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1503 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000db0be0, 0xc00022ee00, 0x5f21738, 0xc000103e00, 0xc001120c00)
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:843 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /home/weiyuping/work/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:841 +0x1fd

Error: The terraform-provider-azurerm plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Comment by ArcturusZhang** [comment]
Hi @jhendrixMSFT here is the code we are using for building the `armcore.Connection`: https://github.com/ArcturusZhang/terraform-provider-azurerm/blob/4aca25d24933919176af7106b24494e6cb560c09/azurerm/internal/common/client_options.go#L60

**Comment by jhendrixMSFT** [comment]
How long does it take to repro, is it immediate or are there several successful polls on the LRO endpoint?

**Comment by jhendrixMSFT** [comment]
Is there any code that replaces the `Body` field on a request as it goes through the pipeline?  It's strange that somehow it changes to an `io.NopCloser`.

**Comment by yupwei68** [comment]
I've sent you the whole log by mail. I think it happens in the first create update.

**Comment by jhendrixMSFT** [comment]
The problem is [here](https://github.com/ArcturusZhang/go-azure-helpers/blob/track2-adoption/policy/logging.go#L27).  The use of [`httputil.DumpRequestOut`](https://golang.org/src/net/http/httputil/dump.go?s=2110:2175#L67) replaces the `Body` which breaks our invariant in the retry policy.

**Comment by jhendrixMSFT** [comment]
Fixed in `sdk/azcore/v0.15.1`
